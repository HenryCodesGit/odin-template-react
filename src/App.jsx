import "./App.css";

import { useState } from "react";

function App() {
  const [heading, setHeading] = useState("Magnificient monkeys");

  const clickHandler = () => {
    setHeading("Radical Rhinos");
  }

  return (<>
    <h1>{heading}</h1>
    <button type="button" onClick={clickHandler}>Click me</button>
  </>);
}

export default App;
