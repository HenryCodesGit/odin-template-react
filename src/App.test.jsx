import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import {describe, it as test, expect} from 'vitest';

import App from './App';

describe("App component", ()=>{

    test("Renders 'magnificient monkeys'", ()=>{
        //Since screen does not have a container property, we can destructure 'render()' to get a container for the tester
        const {container} = render(<App />);
        expect(container).toMatchSnapshot(); //Refer to the snapshot in the 'snapshot' folder
    })

    test("Renders 'radical rhinos' after button click", ()=>{
        async () => {
            const user = userEvent.setup();

                render(<App />);
                //Instead of destructuing, use 'screen' instead.
                
                screen.debug();

                const button = screen.getByRole("button", {name: "Click me"});

                await user.click(button);

                expect(screen.getByRole("heading").textContent).toMatch(/Radical Rhinos/i);
        }

    })

})