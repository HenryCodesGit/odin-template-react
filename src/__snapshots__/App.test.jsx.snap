// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App component > Renders 'magnificient monkeys' 1`] = `
<div>
  <h1>
    Magnificient monkeys
  </h1>
  <button
    type="button"
  >
    Click me
  </button>
</div>
`;
